File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        mlength,i,j = 0,0,1
        my_set = set()
        if len(s)==0:
            return 0
        my_set.add(s[0])
        while i<len(s) and j<len(s):
            while j<len(s) and not s[j] in my_set:
                my_set.add(s[j])
                j+=1
            if j< len(s) and s[j] in my_set:
                mlength = max(mlength,len(my_set))
                while s[i]!=s[j]:
                    my_set.remove(s[i])
                    i+=1
                my_set.remove(s[i])
                i+=1
        return max(mlength,len(my_set))

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        int i = 0,j = 1;
        int mlen = 0;
        Set<Character> set = new HashSet<>();
        if(s.length()==0)
            return 0;
        set.add(s.charAt(0));
        while(j<s.length()){
            while(j<s.length() && !set.contains(s.charAt(j))){
                set.add(s.charAt(j));
                j++;
            }
            if(j<s.length() && set.contains(s.charAt(j))){
                mlen = Math.max(mlen,set.size());
                while(s.charAt(i)!=s.charAt(j)){
                    set.remove(s.charAt(i));
                    i++;
                }
                set.remove(s.charAt(i));
                i++;
            }
        }
        return Math.max(mlen,set.size());
    }
}